Environment Updates

ItsDobes opened the pull request:
Lots of changes -

1. I turned the background from a skybox to a tileset. This is good because now all the pixels in the game are the same size, whereas before the background 'pixels' were much larger than those in the rest of the sprites.

2. I added spikes to replace the 'spikewall' prefab which was just a red rectangle.

3. Added a death animation and SFX

4. Added a lever to our prefabs which allows the player to enable or disable special SwitchBlocks

5. Adjusted the position of several Fliptokens to make the routes to collect them feel more smooth / not impossible

6. Fixed a typo in the former 'PlayerConrollers' folder which says it affected all of Dan's Bookwyrm stuff. I didn't touch anything else in there, just fixed the file name.

7. Copied the Cinemachine camera controller over to Levels 1-9 instead of just having it on Level 10. This broke the pan functionality, not sure how to fix but leaving that for later.

ItsDobes committed:
Added Background to Tileset

The background was formerly a single skybox image, not part of the tileset. I updated all the backgrounds to be included as tilesets. This is good because it ensures that the "pixel size" is consistent across the character model, the background, the object sprites and the environment.

ItsDobes committed:
Added Death Animation, Hazard Tileset and Spikes

Player now has a death animation and sound effect. This will freeze all player physics during death. Added a new tileset for hazards to replace the spikewall prefab. All tiles on the hazard tileset will kill the player. I had to manually adjust all the spikes by 0.25 offset to make them look good attached to walls and floors. I also adjusted the position of a few coins and flip tokens to make them less annoying to collect.

ItsDobes committed:
Adjusted the position of fliptokens

I tested and adjusted the position of some of the more challenging combo-based gems to make the route easier / possible without cheating. (This is just cleaning up some more of the wonkiness that came from the tileset and physics changes).

ItsDobes committed:
Added platforms below all spikes

I added platforms below all the spikes so they aren't just floating midair

ItsDobes committed:
Added a lever that enables/disables SwitchBlocks

I added a lever that makes a sound effect and enables or disables the SwitchBlock objects in the scene when the player touches it. I also reduced the volume of wallsliding because it was annoying, and I changed a single typo in the former 'CharacterConroller' folder and changed it to 'CharacterController' which is why all the Bookwyrm stuff says it was affected. I didn't touch it otherwise.

ItsDobes committed:
Added TMPro and a new typeface for menus

ItsDobes committed:
Updated Main Menu and Level Select Text

Switch all 'Text' objects to TMPro and added a font

ItsDobes committed:
Added foreground and background props to each page

Altered the level design somewhat (esp. pages 2,3,5) to add more design flair that incorporates foreground and background tiles. Improved the SwitchBlockController so it can no longer flip into the position it was already in.

ItsDobes committed:
Updated Level Designs

Added eye candy, fixed the position of gems and other objects, added additional challenges with levers, added a consistent 'underground' look to all the lower terrains, and fixed a bug that prevented loading Level 6 from the overworld menu.

ItsDobes committed:
Added tutorial signs

Tutorial signs explain basic elements of the gameplay

DanielMorin44 commented on Assets/Scripts/Interactables/LeverController.cs:
```diff
@@ -0,0 +1,51 @@
+        }
+    }
+
+    private void PullLever()
```
I don't think PullLever or LiftLever are ever used. If not, lets remove this.

ItsDobes replied:
Fixed!

DanielMorin44 commented on Assets/Scripts/Interactables/TooltipController.cs:
```diff
@@ -0,0 +1,32 @@
+using TMPro;
```
Nitpick, but this class doesn't need TMPro

ItsDobes replied:
Fixed!

ItsDobes committed:
Cut unnecessary code

Cut a few lines of code in LeverController and TooltipController to comply with PR comments
